crates/gitbutler-tauri/src/open.rs

```diff
@@ -0,0 +1,69 @@
+use crate::error::Error;
+use anyhow::anyhow;
+use regex::Regex;
+use std::env;
+use tracing::instrument;
+
+pub fn open_that(path: &str) -> Result<(), Error> {
+    let re = Regex::new(r"^((https://)|(http://)|(mailto:)|(vscode://)|(vscodium://)).+").unwrap();
+    if !re.is_match(path) {
+        return Err(anyhow!("Invalid path format").into());
+    }
+
+    fn clean_env_vars(var_names: &[&str]) -> Vec<(String, String)> {
+        let mut cleaned_vars = Vec::new();
+        for var_name in var_names {
+            if let Ok(var_value) = env::var(var_name) {
+                let cleaned_value = var_value
+                    .split(':')
+                    .filter(|path| !path.contains("appimage-run"))
+                    .collect::<Vec<_>>()
+                    .join(":");
+                cleaned_vars.push((var_name.to_string(), cleaned_value));
+            }
+        }
+        cleaned_vars
+    }
+
+    std::thread::spawn({
```

Comment:
I would remove this behaviour - the function should block and be able to report an error if all commands failed to open the URL.

The non-blocking portion is handled when called through Tauri - it automatically launches it on a thread, and the UI can always deal with that using its own promises.

Correct error handling is definitely preferable here.

Reply:
Sounds good, will clean this up too. 

Just out of curiosity, Is it always the case that tauri handles the non-blocking stuff, or only when annotated with the async keyword in the tauri command macro?

Reply:
It's correct, it's only with the `async` keyword in the command macro. Otherwise it would be blocking.
And now that you are asking, I don't see that macro anywhere and wasn't aware that any function can be exposed like apparently is happening here 😅.

Another thing I apparently missed - I think my lesson here is to not do armchair reviews, I need the IDE to really understand the whole context of the code.

Reply:
Haha isn't it at the bottom of the file here or am I missing something? I just remember reading (was it you and kiril?) someone mention the async keyword in the macro thing, so I wanted to be sure, becuase it's currently missing there. 

I'll add the async keyword and remove this threading 👌

Reply:
Now that I have opened this PR in my IDE it's clear that… the macro exists but is indeed missing the `async` keyword.

I am surprised about the `unwrap()` there though.